tests/parsers/test_pw.py

```diff
@@ -937,3 +937,25 @@ def test_pw_vcrelax_failed_not_converged_nstep(
     assert 'output_parameters' in results
     assert 'output_structure' in results
     assert 'output_trajectory' in results
+
+
+def test_magnetic_moments_v68(
+    fixture_localhost, generate_calc_job_node, generate_parser, generate_inputs, data_regression
+):
+    """Test the parsing of the magnetic moments in QE v6.8."""
+    name = 'magnetic_moments_v68'
+    entry_point_calc_job = 'quantumespresso.pw'
+    entry_point_parser = 'quantumespresso.pw'
+
+    # By setting the `without_xml` option the parsing can only use the stdout content
+    inputs = generate_inputs(calculation_type='relax', metadata={'options': {'without_xml': True}})
+    node = generate_calc_job_node(entry_point_calc_job, fixture_localhost, name, inputs)
+    parser = generate_parser(entry_point_parser)
+    results, calcfunction = parser.parse_from_node(node, store_provenance=False)
+
+    assert calcfunction.is_finished_ok, calcfunction.exit_message
+    assert 'output_trajectory' in results
+
+    data_regression.check({
+        'output_trajectory': results['output_trajectory'].attributes,
```

Comment:
All things good! Just one question out of curiosity: we only check the `output_trajectory.attributes`, but not the parsed actual data of the attributes? sth. like checking `output_trajectory.get_array('atomic_magnetic_moments')`?

Reply:
It's a fair question! I figured that the main point is to see if the parsing fails or not, and as long as the size of the arrays returned is still correct, we're probably in good shape. Also, numpy arrays are a problem for the regression check, it seems:

```
FAILED test_pw.py::test_magnetic_moments_v68 - yaml.representer.RepresenterError: ('cannot represent an object', array([[7.9805, 1.2176]]))
```

Probably because they aren't readily serialisable into YAML format. Of course, we can convert them into lists:

```Python
    data_regression.check({
        'atomic_charges': results['output_trajectory'].get_array('atomic_charges').tolist(),
        'atomic_magnetic_moments': results['output_trajectory'].get_array('atomic_magnetic_moments').tolist(),
    })
``` 

And that works fine. So I'll use that instead, will also add less lines. 😁